### DISCUSSION TITLE
Support export for enums

### DISCUSSION THREAD

**Discussion by phh** [filamentphp/filament#11079] [Ideas]
Hi there!

Today we had a problem when exporting a field, since it was cast to an enum. The fix was easy, just add a `usingState` and jobs done!

However, since the enum was already using the `HasLabel` contract maybe it could just resolve the enum with `getLabel()` if that was returned from that field?

**Comment by LucasSouzaa97** [comment]
Can you show me where to use usingState? I'm having the same problem but I can't solve it

**Reply by phh** [reply]
Can't really remember or find the related code. However, I guess it would be something like:

```php
Column::make('my_enum')
    ->getStateUsing(fn (SomeModel $record) => $record->my_enum->getLabel());
```

**Reply by LucasSouzaa97** [reply]
Thank you very much, from the bottom of my heart, phh! You really saved me here 😁👊🏾

**Reply by nnnnnnnngu** [reply]
You can use the `enum.value` syntax if the enum value (not a label) itself is acceptable for you

```php
Column::make('my_enum.value')

**Reply by ahmetozrahat** [reply]
@phh thank you very much, you saved me a lot of time!

**Reply by phh** [reply]
Thanks for coming back! Guess this could be worth a PR...

**Reply by a-coding-a** [reply]
We should add this into the Enum core concepts docs so it's obvious
